### ISSUE TITLE
Mod not working correctly in Multiplayer for me

### ISSUE THREAD

**Issue by DerTikTak** [loki79uk/FS25_UniversalAutoload#11]
Hey, 

I can edit the loadingzones but i cant load any pallets on my trailer. 
Even the options in the menu on the top left corner are missing

**Comment by loki79uk** [comment]
You need to add the configurations to your server manually.  I wish I could make this issue a sticky:  https://github.com/loki79uk/FS25_UniversalAutoload/issues/8

**Comment by DerTikTak** [comment]
I am so sorry then. Maybe write it down in the readme file. 
What you mean with configurations

**Comment by DerTikTak** [comment]
Ah sorry, now i understand it. I will try it. 
Thanks for your work mate

**Comment by DerTikTak** [comment]
Sadly it looks like its not working for my nitrado server

**Comment by DerTikTak** [comment]
Yeah but only the "mod"-Folder. I have already contacted support, but they claim that they do not have access to the “modSettings” folder

**Comment by loki79uk** [comment]
I'd guess a week or two, because it feels like I have 100 things on my plate at the moment.  There isn't much point getting MP to work when there are still bugs in SP, and all the code changes for that could affect what is needed for MP.

**Comment by DerTikTak** [comment]
I am also an developer and a absolutly understand your thoughts. I thought about do it by my self but never worked with lua so far

**Comment by loki79uk** [comment]
I also need to design a menu for the shop, and depending what buttons are there will affect what should happen with saving settings.  I might be able to get the server to write settings files fairly easily though - I'll have a look when I complete some other things and update here.

**Comment by DerTikTak** [comment]
Would be pretty cool. @ the moment Nitrado wont get any access to other directorys to users. Would appreciate it so much

**Comment by JasonX2001** [comment]
Hi, i add 4NetPlayers to the list. So mod is not usable on public hosters because of lack to get the modsettings folder to show in ftp. 
Mybe the support will change it for us i don't know. Their work shift has ended today, so i will know it tomorrow.

**Comment by DerTikTak** [comment]
Ouh, a second Jason in this conversation :D 
No worrys just take your time, Things you need to do fast wont be perfect tbh

**Comment by r45br** [comment]
Stop the server. Access ftp server. Create a folder named modSettings. Copy the UniversalAutoload.xml file to the created folder. Start the server. I use the 4Netplayers and it worked. Congratulations on the mod.

**Comment by JasonX2001** [comment]
> Temporary solution. Enter career mode and buy all the trailers you want to autoload. Exit career mode. Stop the server completely (in the administrative panel of the contracted server). Access ftp. Create a folder named modSettings. Copy the UniversalAutoload.xml file located in the folder on your computer to the created folder. Start the server. I use the 4Netplayers and it worked. Congratulations on the mod.

Just how? 
I exactly did the same yesterday, just in case it could work but nothing happened. Even restarting the server. It did not show any triggeres to press in the UI top left. I will try it today again, but these where the steps i took. Did you buy the trailers AFTER the XML Upload or did the already bought trailers got to work?

**Comment by DerTikTak** [comment]
> Temporary solution. Enter career mode and buy all the trailers you want to autoload. Exit career mode. Stop the server completely (in the administrative panel of the contracted server). Access ftp. Create a folder named modSettings. Copy the UniversalAutoload.xml file located in the folder on your computer to the created folder. Start the server. I use the 4Netplayers and it worked. Congratulations on the mod.


Thats not working for me. 
I only can enter the "mods" folder with ftp on NItrado

**Comment by loki79uk** [comment]
I had a thought that maybe I could add console commands for this (for now). Something like "save config to server" when in the shop. 

The main issue is how do we even want these to be stored on the server in the final version?  If two players have different settings in MP, do we keep different configurations for each player on the server? Someone has to log in as admin? We can still have multiple admins.  It sounds like could be a nightmare. In the past it has just been a single config for the server and everyone uses the same one, but the default file was created by the mod at launch.

**Comment by DerTikTak** [comment]
That would be a temporary solution for now. It would be enough to log in as admin and then create the file, which everyone would then use in exactly the same way. We are currently not interested in adjusting the loading zones, so we all use the same settings

**Comment by Freddy844** [comment]
> Can you FTP to your server?

I can ftp on my Server Can you say me how i put the autoload in my Server i Place IT on my mod Folder but no Cars have IT  no function in console say autoload disabled

**Comment by loki79uk** [comment]
> That would be a temporary solution for now. It would be enough to log in as admin and then create the file, which everyone would then use in exactly the same way. We are currently not interested in adjusting the loading zones, so we all use the same settings. So one config would currently be the solution in my case

That would be fine to start with, but what about when someone buys a new vehicle that is not yet confgiured?  and then when someone else buys the same one as someone else, etc..

**Comment by JasonX2001** [comment]
> > That would be a temporary solution for now. It would be enough to log in as admin and then create the file, which everyone would then use in exactly the same way. We are currently not interested in adjusting the loading zones, so we all use the same settings. So one config would currently be the solution in my case
> 
> That would be fine to start with, but what about when someone buys a new vehicle that is not yet configured? and then when someone else buys the same one as someone else, etc..

Yeah, imho the solution for now would be the hosters to grant access to that folder over FTP. I don't know a single reason why they at all limit it, because the folder is in your own instance, so you could not mess with other servers. I am still waiting for feedback of the support. 
I realy appreciate you trying your best, to get a temporary solution from your side for us.

**Comment by DerTikTak** [comment]
> > That would be a temporary solution for now. It would be enough to log in as admin and then create the file, which everyone would then use in exactly the same way. We are currently not interested in adjusting the loading zones, so we all use the same settings. So one config would currently be the solution in my case
> 
> That would be fine to start with, but what about when someone buys a new vehicle that is not yet configured? and then when someone else buys the same one as someone else, etc..

They need to contact the admin who need to configure it then in my opinion. 
This way you have one instance that edits and sets the config for all users. This also ensures that nobody exploits the whole thing or uses bugs
Also an temp solution

**Comment by DerTikTak** [comment]
> > They need to contact the admin who need to configure it then in my opinion. This way you have one instance that edits and sets the config for all users. This also ensures that nobody exploits the whole thing or uses bugs. As long as the mod is not finished, I think this solution makes the most sense
> 
> You could have multiple people logged in as admin though, but I agree they just have to know not to overwrite settings. If I can make a function to upload local config to server, and another one to retrieve config from server, both on console command for now I think that will always be useful whatever happens with the final design.

That's exactly it. There just has to be one admin who sets it up and the rest are simply not allowed to change anything.

**Comment by DerTikTak** [comment]
If i could do something for you, let me know

**Comment by deityxox** [comment]
> Thanks,
> 
> I got it to work. I did unterstand it wrong. Not stop the gameserver in the admin panel of the game itself. Stop it from your Serveroverview in 4NetPlayers Backend.

@JasonX2001 


Is the mod currently working on the 4netplayers server? I did the same thing, but when buying a new trailer, the mod appears in the market, but it doesn't work after buying it and getting into the vehicle.

**Comment by deityxox** [comment]
@r45br 
> Every time you have to stop through the overview you need to do the procedure again (copy the folder).


What you mean is that every server is closed and opened, right?

**Comment by JasonX2001** [comment]
> @r45br
> 
> > Every time you have to stop through the overview you need to do the procedure again (copy the folder).
> 
> What you mean is that every server is closed and opened, right?

He means, every time you have to stop the server completly. Not in the Giants Server-Admin-Panel but in the 4NetPlayers Server-Overview. And if its stopped you can make the changes to the Autoload xml.

**Comment by JasonX2001** [comment]
> > Thanks,
> > I got it to work. I did unterstand it wrong. Not stop the gameserver in the admin panel of the game itself. Stop it from your Serveroverview in 4NetPlayers Backend.
> 
> @JasonX2001
> 
> Is the mod currently working on the 4netplayers server? I did the same thing, but when buying a new trailer, the mod appears in the market, but it doesn't work after buying it and getting into the vehicle.

Yes. Playes the whole day with the mod. Tomorrow morning it does its restart, then we will see if it stays or is gone.

**Comment by loki79uk** [comment]
I think the next release will be the opposite to this one, as in every person that saves a vehicle will overwrite the version on the server.  This is still only temporary until I sort out admin rights and such, but it will be a good test none the less (and should solve your server access issue for now).

**Comment by DerTikTak** [comment]
You are my hero. Thanks for your work and time

**Comment by r45br** [comment]
> > > Obrigado, 
> > > consegui fazer funcionar. Eu entendi errado. Não pare o servidor de jogo no painel de administração do jogo em si. Pare-o na sua Serveroverview no 4NetPlayers Backend.
> > 
> > 
> > @JasonX2001
> > O mod está funcionando no servidor 4netplayers? Eu fiz a mesma coisa, mas ao comprar um trailer novo, o mod aparece no mercado, mas não funciona depois de comprá-lo e entrar no veículo.
> 
> Sim. Joga o dia todo com o mod. Amanhã de manhã ele reinicia, então veremos se ele fica ou some.

The configuration will probably disappear.

**Comment by loki79uk** [comment]
> You are my hero. Thanks for your work and time

Did the new version work at all with your server?  I realised that if the file is not written correctly (see #13) then you can't do anything about it, so I'll at least add a console command to clear all settings.

**Comment by DerTikTak** [comment]
Unfortunately, it does not work as it should. When I get the option to load, it only loads one pallet at a time and it is then lying completely at an angle on my trailer

**Comment by DerTikTak** [comment]
Yeah mp. Any chance to fix it?

**Comment by loki79uk** [comment]
> Yeah mp. Any chance to fix it?

Yes, I am working on it, but hard to say..

In FS22 the solution was this:
https://github.com/loki79uk/FS22_UniversalAutoload/blob/b6640f4dd29e4db4d41f2e261556622b74cc245f/UniversalAutoload.lua#L5472-L5484

That didn't work for 25, and I've changed it to this, but still not updating the positions..
https://github.com/loki79uk/FS25_UniversalAutoload/blob/8b02dad63fc1bf4c01b32b35ce47f6dfa102c64f/UniversalAutoload.lua#L5333-L5346

**Comment by DerTikTak** [comment]
And once again I realize how “close to languages” the programming language is :D 
Just take your time

**Comment by loki79uk** [comment]
> And once again I realize how “close to languages” the programming language is :D Just take your time

I've just got it to work!  New release incoming, but have some other things to tidy up first.

**Comment by loki79uk** [comment]
https://github.com/loki79uk/FS25_UniversalAutoload/releases

**Comment by Zanto1608** [comment]
> Cela ne fonctionne pas actuellement pour les arbres/bûches, n'est-ce pas ?

Log trailers (can be configured, but do not load anything yet)
